Add job-processing and marketing material.  Handle multi-step jobs.

manthey opened the pull request:
This adds a react component that shows known tasks with controls based on the yml file.  Input and outputs can have titles and notes.  Jobs can be chained together.

This expects inputs to be in a publicly-readable folder called "OSUMO Inputs".  The current user needs to have read/write access to a folder called "OSUMO Outputs".  If no user is logged it, a fallback user can be used, whose username is expected to be "osumopublicuser".  This fallback user must have read access to OSUMO Inputs and read/write access to OSUMO outputs.

There are three jobs currently defined, one of which is in two stages.

Original comment:

This adds a simple react component that shows known tasks.  When a task is selected, the appropriate input fields are shown.  Currently, there are two hard-coded IDs: one for the folder where source items are located, and one for the folder where data will be written.

I've changed the job spec yml to provide information necessary for the UI.

I've altered how job specs are imported so that they can be enumerated via the GET osumo/tasks endpoint.

I've altered the POST osumo endpoint so that it can process arbitrary tasks.

Progress and results are shown simplistically.

manthey committed:
Adding job processing to the react web pages.

This adds a simple react component that shows known tasks.  When a task is selected, the appropriate input fields are shown.  Currently, there are two hard-coded IDs: one for the folder where source items are located, and one for the folder where data will be written.

I've changed the job spec yml to provide information necessary for the UI.

I've altered how job specs are imported so that they can be enumerated via the GET osumo/tasks endpoint.

I've altered the POST osumo endpoint so that it can process arbitrary tasks.

manthey commented:
@opadron If you want to run this against a local girder instance, you'll need to change the two hard-coded ids in web-external/src/components/process-data.jsx.

manthey committed:
Updated the survival example to show some output.

manthey committed:
Modified so that the surv task outputs a PNG and we show it.

One minor remaining issue is that the temporary file created as part of the task is not eliminated (it would be better if it were in the girder_worker job directory).

manthey committed:
Use the girder worker temp directory for the temp png file.

Change the css styling back to what Girder uses for the footer (it was hiding some data).

opadron commented on web-external/src/style/full-viewport.styl:
```diff
@@ -1,24 +1,23 @@
-  padding-bottom: 88px
+DISABLE#g-app-body-container
+  height 95%
+  padding-bottom 88px
```
What exactly is getting hidden?  Is there a way that we can get the footer bottom-aligned without this issue?  My CSS-foo is failing me.

manthey replied:
If the contents of the main panel are too long, they get hidden behind the footer (and I can't scroll to see them).  It might be easier to show you.

This changed what Girder does -- girder has a minimum content size that the footer appears below.  If the content is longer, the footer is off screen initially, and only visible when you scroll to it).  Do you want to permanently consume screen real-estate for the footer instead?

opadron replied:
I was going for a middle body area that takes the majority of the screen space and scrolls with the content.  In your example case, I would have expected a scroll bar to appear that would scroll the middle body contents while keeping the header/footer/nav bar fixed.

opadron replied:
(and to do so in such a way as to make all body content viewable at the right scroll position).  I thought the `padding-bottom` attribute should have taken care of the footer area?  Am I missing something?  Is the padding just not enough?

manthey replied:
That will work, but we have an @media CSS rule that overrides the app-body-container style you had added (so the margin was reset).

As a design choice, I prefer the footer that doesn't stick to the bottom of the window, as it eats up space where we can be displaying data, and nothing we have in the footer is a common UI action)

opadron replied:
In that case, we can 86 my attempted CSS hack and stick with what's in Girder.

manthey committed:
Removed some css changes that diverged from Girder.

manthey commented:
@opadron For the short term, the hard-coded ids could be removed by creating folders in the current user's public folder if they don't yet exist and using those (named something like "Sumo Source Data" and "Sumo Job Results".  Then this branch would work generically (assuming you are logged in to Girder).

manthey committed:
Merge branch 'master' into job-runner

manthey committed:
Only process our own jobs.

manthey committed:
Merge branch 'master' into job-runner

Conflicts:
	web-external/src/components/header.jsx
	web-external/src/main.jsx
	web-external/src/utils/cookie.jsx
	web-external/src/utils/rest-requests.jsx

manthey committed:
Adjust to semistandard formatting.

manthey committed:
Remove hard coding of folder IDs.

manthey committed:
Add initial iGPSe task.

manthey committed:
Improve how files are stored with jobs.

Before, it was possible to clobber some job information due to the asynchronous nature of jobs.

manthey committed:
Partial work in getting the iGPSe task working.

Still some issues with the parallel sets code, and the buttons for parallel sets don't do anything yet.

manthey committed:
Make the parallel sets widget a component.

Make it work (mostly).  The PLOT button doesn't do anything yet.

manthey committed:
Added second part of iGPSe task.

The tasks don't automatically chain together, yet.

Made it so we can show a list of items, but send the appropriate file to the job instead.  Otherwise, if the item and file are not named the same, the girder worker plugin downloads the item as a directory, but expects to get a file.  I think that this could (should?) be changed in the girder worker plugin, as for a one-file item, I don't see the desired default being the current behavior.

manthey committed:
Initial multi-task implementation.

Ugly, ugly code in the process-data component.

manthey committed:
Fix permissions.  A user must have read access on the input folder and write access on the target folder.

Fix parallel sets so that links and nodes can be toggled.

Better report job errors.

manthey committed:
Updated iGPSe jobs and parallel set widget to produce actual data.

manthey committed:
Sort input files based on whether they match a pattern or not.

Update iGPSe job to hide unused parameters.

Update iGPSe and surv jobs to specify name matches.

manthey committed:
Allow running jobs with a fall-back user.

manthey committed:
Add axes labels, titles, and a legend to the iGPSe graph.

Renamed 'subgroup' to 'group' for consistency.

manthey committed:
Fix a computation that affected the link placement on the parallel set widget.

If two nodes had the same height, one or more links would be misplaced.

manthey committed:
Many css adjustments.

Show notes and display names from tasks.

Fix some typos.

manthey committed:
Have a restriction of files by name as an input option.

Improve legends of the survival plot.

Fix some errors in parallelSets.

Change the background of the header.

manthey committed:
Change bottom links to github.com/osumo/osumo

waxlamp committed:
Import main marketing content from github.io site

waxlamp committed:
Import modal content as permanent content in main page

waxlamp committed:
Remove "Home" heading

waxlamp committed:
Remove "thumbnails" of projects from project section

manthey committed:
Hide left-side links that aren't finished.

waxlamp committed:
Add "SUMO" heading to top of marketing content

waxlamp committed:
Remove Kitware copyright notice

waxlamp committed:
Fix typo in phone number

waxlamp committed:
Remove all modal specifiers

waxlamp committed:
Use fluid containers

waxlamp committed:
Use the full column width for all text

waxlamp committed:
Use h3 headers for project titles

waxlamp committed:
Do not use justified text

waxlamp committed:
Center the youtube iframes

waxlamp committed:
Remove all empty class specified

waxlamp committed:
Remove redundant div tags

Commit:
bump webpack version

Commit:
fix - process is no longer a module

Commit:
Merge pull request #9 from osumo/marketing

Marketing materials import
